sed/core/processor.py

```diff
@@ -178,10 +178,35 @@ def __repr__(self):
             df_str = "Data Frame: No Data loaded"
         else:
             df_str = self._dataframe.__repr__()
-        attributes_str = f"Metadata: {self._attributes.metadata}"
-        pretty_str = df_str + "\n" + attributes_str
+        pretty_str = df_str + "\n" + "Metadata: " + "\n" + self._attributes.__repr__()
         return pretty_str
 
+    def _repr_html_(self):
+        html = "<div>"
+
+        html += (
+            f"<details><summary>Dataframe</summary>{self.dataframe.head()._repr_html_()}</details>"
+        )
```

Comment:
Computing the head sometimes takes a substantial amount of time (up to 30 s with the tutorial2). I would not do this per default, as the repr should be fast, I would say.

Reply:
Good point. Was just an initial idea to choose between dask or head.